lms/djangoapps/commerce/tests.py

```diff
@@ -0,0 +1,247 @@
+                sku=uuid4().hex.decode('ascii')
+            )
+
+    def test_login_required(self):
+        """
+        The view should return HTTP 403 status if the user is not logged in.
+        """
+        self.client.logout()
+        self.assertEqual(403, self._post_to_view().status_code)
+
+    @data('delete', 'get', 'put')
+    def test_post_required(self, method):
+        """
+        Verify that the view only responds to POST operations.
+        """
+        response = getattr(self.client, method)(self.url)
+        self.assertEqual(405, response.status_code)
+
+    def test_invalid_course(self):
+        """
+        If the course does not exist, the view should return HTTP 406.
+        """
+        # TODO Test inactive courses, and those not open for enrollment.
+        self.assertEqual(406, self._post_to_view('aaa/bbb/ccc').status_code)
+
+    def test_invalid_request_data(self):
+        """
+        If invalid data is supplied with the request, the view should return HTTP 406.
+        """
+        self.assertEqual(406, self.client.post(self.url, {}).status_code)
+        self.assertEqual(406, self.client.post(self.url, {'not_course_id': ''}).status_code)
+
+    @httpretty.activate
+    @data(400, 401, 405, 406, 429, 500, 503)
+    def test_ecommerce_api_bad_status(self, status):
+        """
+        If the E-Commerce API returns an HTTP status not equal to 200, the view should log an error and return
+        an HTTP 503 status.
+        """
+        self._mock_ecommerce_api(status=status, body=json.dumps({'user_message': 'FAIL!'}))
+        response = self._post_to_view()
+        self.assertValidEcommerceApiErrorResponse(response)
+        self.assertFalse(CourseEnrollment.is_enrolled(self.user, self.course.id))
+
+    @httpretty.activate
+    def test_ecommerce_api_timeout(self):
+        """
+        If the call to the E-Commerce API times out, the view should log an error and return an HTTP 503 status.
+        """
+        # Verify that the view responds appropriately if calls to the E-Commerce API timeout.
+        def request_callback(_request, _uri, _headers):
+            """ Simulates API timeout """
+            raise Timeout
+
+        self._mock_ecommerce_api(body=request_callback)
+        response = self._post_to_view()
+        self.assertValidEcommerceApiErrorResponse(response)
+        self.assertFalse(CourseEnrollment.is_enrolled(self.user, self.course.id))
+
+    @httpretty.activate
+    def test_ecommerce_api_bad_data(self):
+        """
+        If the E-Commerce API returns data that is not JSON, the view should return an HTTP 503 status.
+        """
+        self._mock_ecommerce_api(body='TOTALLY NOT JSON!')
+        response = self._post_to_view()
+        self.assertValidEcommerceApiErrorResponse(response)
+        self.assertFalse(CourseEnrollment.is_enrolled(self.user, self.course.id))
+
+    @data(True, False)
+    @httpretty.activate
+    def test_course_with_sku(self, user_is_active):
```

Comment:
This could be more accurately named: `test_course_with_honor_seat_sku`. Not all requests to the ecommerce API should result in an enrollment.